plugins/inputs/marathon/marathon.go

```diff
@@ -0,0 +1,162 @@
+package marathon
+
+import (
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	jsonparser "github.com/influxdata/telegraf/plugins/parsers/json"
+)
+
+type Marathon struct {
+	Servers     []string
+	MetricTypes []string `toml:"metric_types"`
+}
+
+var allMetricTypes = []string{
+	"gauges", "counters", "histograms", "meters", "timers",
+}
+
+var sampleConfig = `
+  ## A list of Marathon servers.
+  servers = ["localhost:8080"]
+  ## Metric types to be collected, by default all enabled.
+  metric_types = [
+    "gauges",
+    "counters",
+    "histograms",
+    "meters",
+    "timers",
+  ]
+`
+
+// SampleConfig returns a sample configuration block
+func (m *Marathon) SampleConfig() string {
+	return sampleConfig
+}
+
+// Description just returns a short description of the Marathon plugin
+func (m *Marathon) Description() string {
+	return "Telegraf plugin for gathering metrics from Marathon hosts"
+}
+
+func (m *Marathon) SetDefaults() {
+	if len(m.MetricTypes) == 0 {
+		m.MetricTypes = allMetricTypes
+	}
+}
+
+// Gather() metrics from given list of Marathon servers
+func (m *Marathon) Gather(acc telegraf.Accumulator) error {
+	var wg sync.WaitGroup
+	var errorChannel chan error
+
+	m.SetDefaults()
+
+	errorChannel = make(chan error, len(m.Servers))
+
+	for _, v := range m.Servers {
+		wg.Add(1)
+		go func(c string) {
+			defer wg.Done()
+			errorChannel <- m.gatherMetrics(c, ":8080", acc)
```

Comment:
You can simplify the code, and make the errors logs easier to read by doing `acc.AddError(...)`.